Implement search filter changed indicator

PikachuEXE opened the pull request:
**Pull Request Type**
Please select what type of pull request this is:
- [X] Feature Implementation

**Related issue**
#1251 

**Description**
When search filter is updated and value is different from default
Update UI to indicate that

**Screenshots (if appropriate)**
Actual effect to be decided

![Screen Shot 2021-05-07 at 11 55 35 ](https://user-images.githubusercontent.com/1018543/117400121-96514200-af34-11eb-857a-e3ef9396bdd9.png)
![Screen Shot 2021-05-07 at 11 55 38 ](https://user-images.githubusercontent.com/1018543/117400131-99e4c900-af34-11eb-9e4f-a8f63de1b373.png)


**Testing (for code that is not small enough to be easily understandable)**
- Open filter panel
- Change any filter from default value and back
- Observe visual change

**Desktop (please complete the following information):**
 - OS: MacOS
 - OS Version: 10.15.7
 - FreeTube version: 12.0.0 (latest on `development` actually)

PikachuEXE commented:
Here is how it looks in different colors

![Screen Shot 2021-05-12 at 11 47 38 ](https://user-images.githubusercontent.com/1018543/117915479-f15dad00-b317-11eb-8f08-0634b9f6f2ec.png)
![Screen Shot 2021-05-12 at 11 47 44 ](https://user-images.githubusercontent.com/1018543/117915483-f3c00700-b317-11eb-84ee-e65efffaccfa.png)
![Screen Shot 2021-05-12 at 11 47 48 ](https://user-images.githubusercontent.com/1018543/117915484-f4589d80-b317-11eb-9531-cbf0a55034e3.png)
![Screen Shot 2021-05-12 at 11 47 53 ](https://user-images.githubusercontent.com/1018543/117915486-f4f13400-b317-11eb-9247-35cdb47eb5a2.png)
![Screen Shot 2021-05-12 at 11 47 57 ](https://user-images.githubusercontent.com/1018543/117915488-f589ca80-b317-11eb-88d0-f071acc30610.png)
![Screen Shot 2021-05-12 at 11 48 04 ](https://user-images.githubusercontent.com/1018543/117915490-f589ca80-b317-11eb-9da1-3afe9731e341.png)

PikachuEXE commented:
I think it's better to use secondary color theme for the effect

![Screen Shot 2021-05-12 at 13 14 46 ](https://user-images.githubusercontent.com/1018543/117921904-23750c00-b324-11eb-9a2e-95b5fbbf628e.png)
![Screen Shot 2021-05-12 at 13 14 50 ](https://user-images.githubusercontent.com/1018543/117921910-25d76600-b324-11eb-896d-9647f860264c.png)
![Screen Shot 2021-05-12 at 13 14 55 ](https://user-images.githubusercontent.com/1018543/117921911-27089300-b324-11eb-8ca8-0cb68560660d.png)
![Screen Shot 2021-05-12 at 13 15 04 ](https://user-images.githubusercontent.com/1018543/117921912-27a12980-b324-11eb-98d1-d4d375412336.png)

efb4f5ff-1298-471a-8973-3d47447115dc commented:
In my opinion it is better to go with the primary color. When u open the filter window u will see that the color of the filters matches the primary color, because of that i think it would be more recognizable for the user if the color is the same as the main color, because it is a familiar color. 

In short the primary color would look more uniform and recognizable to the user.

PikachuEXE commented:
Changed back to use primary color

GilgusMaximus commented:
Okay, thanks taking on this. I finally had some time to test it out and it works great.
However three things are required to merge this:

1) We need a solution for when the user enables the setting 'Match Top Bar with Main Color'. Because then you do not see whether the circle is actually displayed, cause it is the same color as the top bar. This could be, that the ring is now displayed in the same color as the other icons in the top bar. They change from white to dark, so a similar switch for the ring should be doable.

2) The filter icon should be moved only a bit to the right, or the ring's radius should be reduced, so it is not cut of by the search box on it's left hand side.

3) The linter issues in your files must be resolved, otherwise the linter will complain all the time. Which is very annoying

But other than that, I really like the solution.

Best regards,
Gilgus

PikachuEXE commented:
1. Updated to handle "Match Top Bar with Main Color". See screenshots below.
2. Moved the button via `margin-left` and also reduced effect radius 20px > 10px
3. Linter warnings resolved
![Screen Shot 2021-05-25 at 13 43 44 ](https://user-images.githubusercontent.com/1018543/119445875-14e51680-bd60-11eb-9c27-1438349967fb.png)
![Screen Shot 2021-05-25 at 13 43 49 ](https://user-images.githubusercontent.com/1018543/119445876-157dad00-bd60-11eb-9150-f8eb676ff131.png)
![Screen Shot 2021-05-25 at 13 44 27 ](https://user-images.githubusercontent.com/1018543/119445877-16164380-bd60-11eb-809b-4325f38bb025.png)
![Screen Shot 2021-05-25 at 13 44 30 ](https://user-images.githubusercontent.com/1018543/119445878-16aeda00-bd60-11eb-94fe-85ee91e9d4a6.png)

efb4f5ff-1298-471a-8973-3d47447115dc commented:
Nice solution for the match with top bar color

GilgusMaximus commented:
Thanks again for making the changes.
Really like how it looks now.

Gonna merge this :)
